modules/stochastic_tools/src/multiapps/SamplerFullSolveMultiApp.C

```diff
@@ -67,13 +66,26 @@ SamplerFullSolveMultiApp::SamplerFullSolveMultiApp(const InputParameters & param
   }
   else
     init(_sampler.getNumberOfRows());
+
+  _number_of_sampler_rows = _sampler.getNumberOfRows();
 }
 
 void SamplerFullSolveMultiApp::preTransfer(Real /*dt*/, Real /*target_time*/)
 {
-  // Reinitialize app to original state prior to solve, if a solve has occured
-  if (_solved_once && _mode == StochasticTools::MultiAppMode::NORMAL)
-    initialSetup();
+  if (_mode == StochasticTools::MultiAppMode::NORMAL)
+  {
+    // Reinitialize MultiApp size
+    const auto num_rows = _sampler.getNumberOfRows();
+    if (num_rows != _number_of_sampler_rows)
+    {
+      init(_sampler.getNumberOfRows());
```

Comment:
Do you need to call `initialSetup` after this?

Reply:
I don't think so that is handled by existing calls to initialSetup which now happen during each execution